nerfstudio/data/datamanagers/full_images_datamanager.py

```diff
@@ -547,5 +563,193 @@ def _undistort_image(
         K = undist_K.numpy()
     else:
         raise NotImplementedError("Only perspective and fisheye cameras are supported")
-
+    # print("final:", image.shape, camera.width, camera.height) # prints 'final: (959, 539, 3) tensor([540]) tensor([960])'
     return K, image, mask
+
+
+## Let's implement a parallelized splat dataloader!
+def undistort_idx(idx: int, dataset: TDataset, image_type: Literal["uint8", "float32"] = "float32") -> Dict[str, torch.Tensor]:
+    """Undistorts an image to one taken by a linear (pinhole) camera model and updates the dataset's camera intrinsics to a linear camera model"""
+    data = dataset.get_data(idx, image_type)
+    camera = dataset.cameras[idx].reshape(())
+    # dataset.cameras.width[idx] = data["image"].shape[1]
+    # dataset.cameras.height[idx] = data["image"].shape[0]
+    if idx == 48:
+        # breakpoint()
+        print("beginning", camera.width, camera.height)
+    assert data["image"].shape[1] == camera.width.item() and data["image"].shape[0] == camera.height.item(), (
+        f'The size of image ({data["image"].shape[1]}, {data["image"].shape[0]}) loaded '
+        f'does not match the camera parameters ({camera.width.item(), camera.height.item()}), idx = {idx}'
+    )
+    if camera.distortion_params is None or torch.all(camera.distortion_params == 0):
+        return data
+    K = camera.get_intrinsics_matrices().numpy()
+    distortion_params = camera.distortion_params.numpy()
+    image = data["image"].numpy()
+    K, image, mask = _undistort_image(camera, distortion_params, data, image, K)
+    # print(image.shape[1]) # outputs 539
+    # print(cameras[48].reshape(()).width.item()) # outputs 540
+    data["image"] = torch.from_numpy(image)
+    if mask is not None:
+        data["mask"] = mask
```

Comment:
hmm, doesn't the mask need to be rectified / undistorted too? not sure if the code in `main` does this

Reply:
Yep! the `_undistort_image` function on the main branch returns an undistorted mask